sdk/src/types.rs

```diff
@@ -200,26 +200,25 @@ pub struct HardwareWalletAccount {
 }
 
 impl TransactionSigner for HardwareWalletAccount {
-    fn sign_transaction(&self, txn: RawTransaction) -> SignedTransaction {
-        let signature = self
-            .sign_arbitrary_message(
-                signing_message(&txn)
-                    .expect("Unable to convert txn to signing message.")
-                    .as_ref(),
-            )
-            .expect("Unable to sign transaction from hardware wallet.");
-        SignedTransaction::new(txn, self.public_key().clone(), signature)
+    fn sign_transaction(&self, txn: RawTransaction) -> Result<SignedTransaction> {
+        let signature = self.sign_arbitrary_message(
+            signing_message(&txn)
+                .expect("Unable to convert txn to signing message.")
+                .as_ref(),
+        )?;
+        Ok(SignedTransaction::new(
+            txn,
+            self.public_key().clone(),
+            signature,
+        ))
     }
 
     fn sign_with_transaction_builder(
         &mut self,
         builder: TransactionBuilder,
     ) -> Result<SignedTransaction> {
         let two_minutes = Duration::from_secs(2 * 60);
-        let current_time = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .expect("Failed to get system time")
-            + two_minutes;
+        let current_time = SystemTime::now().duration_since(UNIX_EPOCH)? + two_minutes;
```

Comment:
curios, why do we add two_minutes?

Reply:
Good question! Greg and I discussed about this, 1min might be too short, but 2 mins should be enough for people to pick up their Ledger to review and sign the transaction directly from Ledger